e2e/inspect/inspect.go

```diff
@@ -0,0 +1,182 @@
+			"org.label-schema.usage.singularity.deffile.from": "alpine:latest",
+			"org.label-schema.usage.singularity.runscript.help": "/.singularity.d/runscript.help",
+			"org.label-schema.usage.singularity.version": "3.2.1-660.g4c8a84050"
+		}
+	},
+	"type": "container"
+}`
+
+const expectedRunscriptJson = `
+{
+	"attributes": {
+		"apps": "",
+		"runscript": "#!/bin/sh\n\necho \"E2E Test Container\"\n\n\n"
+	},
+	"type": "container"
+}`
+
+const expectedListAppsJson = `
+{
+	"attributes": {
+		"apps": "hello\nworld\n"
+	},
+	"type": "container"
+}`
+
+const expectedTestJson = `
+{
+	"attributes": {
+		"apps": "",
+		"test": "#!/bin/sh\n\nls /\ntest -d /\ntest -d /etc\n\n\n"
+	},
+	"type": "container"
+}`
+
+const expectedEnvironmentJson = `
+{
+	"attributes": {
+		"apps": "",
+		"environment": {
+			"90-environment.sh": "#!/bin/sh\n#Custom environment shell code should follow\n\n\nexport test=\"testing\"\nexport e2e=\"e2e testing\"\n\n\n"
+		}
+	},
+	"type": "container"
+}`
+
+func runInspectCommand(inspectType string) ([]byte, error) {
+	argv := []string{"inspect", "--json", inspectType, "testdata/test.sif"}
+	cmd := exec.Command("singularity", argv...)
+
+	return cmd.CombinedOutput()
+}
+
+func singularityInspect(t *testing.T) {
+	tests := []struct {
+		name      string
+		insType   string
+		json      []string
+		expectOut string
+	}{
+		{
+			name:      "label E2E",
+			insType:   "--labels",
+			json:      []string{"attributes", "labels", "E2E"},
+			expectOut: expectedLabelsJson,
+		},
+		{
+			name:      "label",
+			insType:   "--labels",
+			json:      []string{"attributes", "labels", "HI"},
+			expectOut: expectedLabelsJson,
+		},
+		{
+			name:      "label",
+			insType:   "--labels",
+			json:      []string{"attributes", "labels", "e2e"},
+			expectOut: expectedLabelsJson,
+		},
+		{
+			name:      "label",
+			insType:   "--labels",
+			json:      []string{"attributes", "labels", "hi"},
+			expectOut: expectedLabelsJson,
+		},
+		{
+			name:      "runscript",
+			insType:   "--runscript",
+			json:      []string{"attributes", "runscript"},
+			expectOut: expectedRunscriptJson,
+		},
+		{
+			name:      "list apps",
+			insType:   "--list-apps",
+			json:      []string{"attributes", "apps"},
+			expectOut: expectedListAppsJson,
+		},
+		{
+			name:      "test",
+			insType:   "--test",
+			json:      []string{"attributes", "test"},
+			expectOut: expectedTestJson,
+		},
+		{
+			name:      "environment",
+			insType:   "--environment",
+			json:      []string{"attributes", "environment", "90-environment.sh"},
+			expectOut: expectedEnvironmentJson,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, test.WithPrivilege(func(t *testing.T) {
+			out, err := runInspectCommand(tt.insType)
+			if err != nil {
+				t.Fatalf("Unexpected failure: %s: %s", string(out), err)
+			}
+
+			// Check the E2E label in test.sif, does it match our expected output
+			v, err := jsonparser.GetString(out, tt.json...)
```

Comment:
this bit here is extracting the object specified by tt.json from out. That's fine.